base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1120 @@
+}
+
+BOOL OnCreate(HWND hwnd)
+{
+    TCITEMW item;
+
+    dv.hTab = GetDlgItem(hwnd, IDC_TAB);
+    dv.hChoicePage = CreateDialogW(GetModuleHandleW(NULL), MAKEINTRESOURCE(IDD_CHOICE_PAGE), hwnd, ChoicePageDlgProc);
+    EnableDialogTheme(dv.hChoicePage);
+    dv.hOptionsPage = CreateDialogW(GetModuleHandleW(NULL), MAKEINTRESOURCE(IDD_OPTIONS_PAGE), hwnd, OptionsPageDlgProc);
+    EnableDialogTheme(dv.hOptionsPage);
+
+    ZeroMemory(&item, sizeof(item));
+    item.mask = TCIF_TEXT;
+    item.pszText = L"Disk Cleanup";
+    (void)TabCtrl_InsertItem(dv.hTab, 0, &item);
+
+    item.pszText = L"More Options";
+    (void)TabCtrl_InsertItem(dv.hTab, 1, &item);
+
+    OnTabWndSelChange();
+
+    return TRUE;
+}
+
+BOOL OnCreateSageset(HWND hwnd)
+{
+    TCITEMW item;
+
+    dv.hTab = GetDlgItem(hwnd, IDC_TAB_SAGESET);
+    dv.hSagesetPage = CreateDialogW(GetModuleHandleW(NULL), MAKEINTRESOURCE(IDD_SAGESET_PAGE), hwnd, SagesetPageDlgProc);
+    EnableDialogTheme(dv.hSagesetPage);
+
+    ZeroMemory(&item, sizeof(item));
+    item.mask = TCIF_TEXT;
+    item.pszText = L"Disk Cleanup";
+    (void)TabCtrl_InsertItem(dv.hTab, 0, &item);
+
+    ShowWindow(dv.hSagesetPage, SW_SHOW);
+    BringWindowToTop(dv.hSagesetPage);
+
+    return TRUE;
+}
+
+void OnTabWndSelChange(void)
+{
+    switch (TabCtrl_GetCurSel(dv.hTab))
+    {
+        case 0:
+            ShowWindow(dv.hChoicePage, SW_SHOW);
+            ShowWindow(dv.hOptionsPage, SW_HIDE);
+            BringWindowToTop(dv.hChoicePage);
+            break;
+
+        case 1:
+            ShowWindow(dv.hChoicePage, SW_HIDE);
+            ShowWindow(dv.hOptionsPage, SW_SHOW);
+            BringWindowToTop(dv.hOptionsPage);
+            break;
+    }
+}
+
+void SelItem(HWND hwnd, int index)
+{
+    DIRECTORIES Dir = index;
+    switch (Dir)
+    {
+        case TEMPORARY_FILE:
+            ShowWindow(GetDlgItem(hwnd, IDC_VIEW_FILES), SW_HIDE);
+            SetDetails(IDS_DETAILS_TEMP, IDC_STATIC_DETAILS, hwnd);
+            break;
+
+        case RECYCLE_BIN:
+            ShowWindow(GetDlgItem(hwnd, IDC_VIEW_FILES), SW_HIDE);
+            SetDetails(IDS_DETAILS_RECYCLE, IDC_STATIC_DETAILS, hwnd);
+            break;
+
+        case OLD_CHKDSK_FILES:
+            ShowWindow(GetDlgItem(hwnd, IDC_VIEW_FILES), SW_HIDE);
+            SetDetails(IDS_DETAILS_CHKDSK, IDC_STATIC_DETAILS, hwnd);
+            break;
+
+        case RAPPS_FILES:
+            ShowWindow(GetDlgItem(hwnd, IDC_VIEW_FILES), SW_SHOW);
+            SetDetails(IDS_DETAILS_RAPPS, IDC_STATIC_DETAILS, hwnd);
+            break;
+    }
+}
+
+long long CheckedItem(int index, HWND hwnd, HWND hList, long long Size)
+{
+    uint64_t TempSize = sz.TempSize;
+    uint64_t RecycleSize = sz.RecycleBinSize;
+    uint64_t DownloadSize = sz.ChkDskSize;
+    uint64_t RappsSize = sz.RappsSize;
+    DIRECTORIES Dir = index;
+    switch (Dir)
+    {
+        case TEMPORARY_FILE:
+            if (ListView_GetCheckState(hList, index))
```

Comment:
It would make more sense to either pass `Dir` here again, or even just `TEMPORARY_FILE`